From d033511e7fdf63f83d29f52db98011bd3c2ab49b Mon Sep 17 00:00:00 2001
From: chengjunlu <chengjun.lu@intel.com>
Date: Wed, 1 Jun 2022 09:16:43 +0800
Subject: [PATCH 19/36] Add the device module to the module list with the name
 of torch.xxx in _register_device_module. The user can use the extension
 device module as a native module in torch namespace. (#50)

---
 torch/__init__.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/torch/__init__.py b/torch/__init__.py
index abd8c1f918..150ce8f1f8 100644
--- a/torch/__init__.py
+++ b/torch/__init__.py
@@ -779,3 +779,5 @@ def _register_device_module(device_type, module):
         raise RuntimeError("The runtime module of '{}' has already "
                            "been registered with '{}'".format(device_type, getattr(m, device_type)))
     setattr(m, device_type, module)
+    torch_module_name = '.'.join([__name__, device_type])
+    sys.modules[torch_module_name] = module
-- 
2.25.1

